q2_types/per_sample_sequences/_transformer.py

```diff
@@ -107,3 +111,85 @@ def _4(dirfmt: CasavaOneEightSingleLanePerSampleDirFmt) \
         -> SingleLanePerSamplePairedEndFastqDirFmt:
     return _single_lane_per_sample_fastq_helper(
         dirfmt, SingleLanePerSamplePairedEndFastqDirFmt)
+
+
+@plugin.register_transformer
+def _5(dirfmt: QIIME1DemultiplexedFastqDirFmt)\
+        -> SingleLanePerSampleSingleEndFastqDirFmt:
+    result = SingleLanePerSampleSingleEndFastqDirFmt()
+
+    out_manifest = FastqManifestFormat()
+    out_manifest_fh = out_manifest.open()
+    out_manifest_fh.write('sample-id,filename,direction\n')
+
+    # keep track of what samples and corresponding read directions have
+    # been observed, to ensure that we only have one read direction per
+    # sample across the input fastq files
+    observed_sample_reads = {}
+    directions = ['forward', 'reverse']
+    in_manifest = iter(dirfmt.manifest.view(QIIME1FastqManifestFormat).open())
+    next(in_manifest)
+
+    for line in in_manifest:
+        if line.startswith('#'):
+            continue
+        filename, read_number, lane_number, phred_offset = \
+            line.strip().split(',')
+        read_number = int(read_number)
+        lane_number = int(lane_number)
+        phred_offset = int(phred_offset)
+        in_fastq_filepath = str(dirfmt.path / filename)
+        data = {}
+        for seq in skbio.io.read(in_fastq_filepath,
+                                 constructor=skbio.DNA,
+                                 format='fastq',
+                                 verify=False,
+                                 phred_offset=phred_offset):
+
+            sample_id = seq.metadata['id'].split('_')[0]
```

Comment:
I think this needs to be parsed in the same way as in my comment [here](https://github.com/qiime2/q2-types/pull/41/files#r78620744)? If so, make a private utility function to centralize that logic